DevEdu.API/DevEdu.DAL/Repositories/CourseRepository.cs

```diff
@@ -70,16 +71,35 @@ public CourseDto GetCourse(int id)
                 commandType: CommandType.StoredProcedure
                 )
                 .FirstOrDefault();
-            return result;
         }
 
         public List<CourseDto> GetCourses()
         {
+            var courseDictionary = new Dictionary<int, CourseDto>();
+            CourseDto result;
             return _connection
-                .Query<CourseDto>(
+                .Query<CourseDto, GroupDto, CourseDto>(
                     _courseSelectAllProcedure,
+                    (course, group) =>
+                    {
+
+                        if (!courseDictionary.TryGetValue(course.Id, out result))
+                        {
+                            result = course;
+                            result.Groups = new List<GroupDto> { group };
+                            courseDictionary.Add(course.Id, result);
+                        }
+                        else
+                        {
+                            result.Groups.Add(group);
+                        }
+
+                        return result;
+                    },
+                    splitOn: "CourseId",
```

Comment:
splitOn must be "Id" everywhere (to be consistent)
so you need some changes in stored procedure